### ISSUE TITLE
[Bug] SponsorBlock Skip Segment not working

### ISSUE THREAD

**Issue by botetta** [YTVanced/Vanced#753]
**Bug description**
In the SponsorBlock settings, you can set it so that a button pops up to Skip segments. This was working perfectly before, but since the latest updates, it's broken, and half the time, the button doesn't show up.

On the progress bar of the video, you can still see the green/purple/etc colors that show that there's something you should be able to skip there, but the button sometimes doesn't appear.

Restarting the app and going to the same video usually fixes this.

**Variant**
Non-Root

**Vanced version**
16.16.38 - Build - 01.51.00

**Android version**
Android 11

**Device**
OnePlus 8T

**Steps to Reproduce**
Steps to reproduce the error:
1. Go to SponsorBlock settings and change the behavior to show Skip button
2. Go on a video which has some skippable segments
3. Go to the skippable segment
4. There's a chance that the skip button won't appear (It seems random when the bug happens, I can't find a clear way to reproduce the bug every time. However it's pretty common, I'd say around 50% of the time this happens. Maybe it happens when you've been using the app for a while without exiting it, because if you restart the app, the first video with SponsorBlock segments will always work properly in my experience)

**Expected behavior**
The skip segment button should appear every time.

**Comment by milindgoel15** [comment]
I tested it personally and I would say it works 100% of times for me. Try to clear data and see. If you could reproduce it with a screen recording, might be helpful

**Comment by botetta** [comment]
https://streamable.com/023gnc

Here's the video. You can see that at first, the button doesn't appear, and after restarting the app and going to the same video, it does appear.

**Comment by milindgoel15** [comment]
Closing this issue due to inactivity. If anyone is facing such issue, Please try reinstalling vanced. Feel free to reopen if anyone's got any useful take on the bug by submitting a video recording (through a better site ofc)

**Comment by botetta** [comment]
https://easyupload.io/j5rntx

Sorry for responding just now. 

I uploaded the video file to this website, you can easily download it and then view it. You can see that at first, the button doesn't appear, and after restarting the app and going to the same video, it does appear.

**Comment by milindgoel15** [comment]
Alright, I already informed the other modder of sponsorblock implementation. Tho main modder who implemented won't come back till June. Hence, we are unsure when this will be fixed. We can't able to reproduce such issue and find the cause. If something comes up, the future updates will have it.

**Comment by botetta** [comment]
Update! 

I have found crucial information about this bug, that should make it easier to figure out the cause and fix the issue.

So I have found out that this issue is directly related to **YouTube shorts**

Basically, sponsorblock stops functioning properly after watching one YouTube short, it is in fact, not _random_.

As an example, if you watch 3 normal YouTube videos, sponsorblock will work on all of them, but if you watch just one YouTube short, sponsorblock stops working properly until you restart the app

Also, some additional information: the skip segment button is not the only thing that stops working after watching a YouTube short (the skip segment button doesn't appear), if you have "Allow adding a new segment" (might not be an exact translation, as my app is not in English) turned on in the settings, this button will also stop functioning, until you restart the app. (There might be other things that break after watching a YouTube short, I'm not sure.) 

This could explain why some people didn't have this problem: some people simply don't use YouTube shorts.

I hope this helps in narrowing down the source of the issue. And yes, the issue persists on the current version, 16.20.35
